api-docs.md

```diff
@@ -561,6 +563,101 @@ There are, however, a few catches to this:
 - Sorting is only applied at the Druid level. Therefore, the results of a sort are not guaranteed to be accurate if Fili 
 performs any post-Druid calculations on one of the metrics that you are sorting on.
 
+Asynchronous Queries
+--------------------
+Fili supports asynchronous data queries. A new parameter `asyncAfter` is added on data queries. The `asyncAfter`
+parameter will control whether a data query should always be synchronous, or transition from synchronous to asynchronous
+ on the fly. If `asyncAfter=never` then Fili will wait indefinitely for the data, and hold the connection with the
+ client open as long as allowed by the network. This will be the default. However, the default behavior of `asyncAfter`
+ may be modified by setting the `default_asyncAfter` configuration parameter. If `asyncAfter=0`, the query is
+ asynchronous immediately.
+
+If it takes less time than the timeout for the data to be sent back, then the results are sent to the client.
+If `asyncAfter=t` for `t` an integer in milliseconds then the query will start synchronously. If the query takes longer
+than `t` milliseconds, then the query becomes asynchronous. If the timeout passes, and the data has not come back, then
+the user receives a `202 Accepted` response and the [job meta-data](#job-meta-data).
+
+### Jobs Endpoint
+The jobs endpoint is the one stop shop for queries about asynchronous jobs. This endpoint is responsible for:
+
+1. Providing the status of tickets queried via the `jobs/TICKET` resource.
+2. Provide a list of all jobs in the system.
+3. Providing access to the results via the `jobs/TICKET/results` resource.
+
+#### GET-ting status of all tickets
+A user may get the status of all tickets by sending a `GET` to `jobs` endpoint.
+
+```
+https://HOST:PORT/v1/jobs
+```
+If no jobs are available in the system, we get an empty collection.
+
+The `jobs` endpoint supports the `filters` query param that filters the JobRows according to the filter condition and
+only returns the JobRows that satisfy the filter condition.
+
+The format of a filter string is:
+```
+jobFieldName-filterOperation[values]
+```
+Multiple comma separated filters are supported and `And`ed by default. Filters on jobs endpoint are similar to
+[filters](#filtering) on data endpoint where jobField name is used to specify the field to filter on.
+
+#### GET-ting Ticket Status
+When the user sends a `GET` request to `jobs/TICKET`, Fili will look up the specified ticket and return the ticket's
+meta-data as follows:
+
+###### Job meta-data
+```json
+{
+    "query": "https://HOST:PORT/v1/data/QUERY",
+    "results": "https://HOST:PORT/v1/jobs/TICKET/results",
+    "syncResults": "https://HOST:PORT/v1/jobs/TICKET/results?asyncAfter=never",
+    "self": "https://HOST:PORT/v1/jobs/TICKET",
+    "status": ONE OF ["pending", "success", "error"],
+    "jobTicket": "TICKET",
+    "dateCreated": "ISO 8601 DATETIME",
+    "dateUpdated": "ISO 8601 DATETIME",
```

Comment:
@cdeszaq Safety, shmafety. I just don't want to interrupt whatever I'm doing in six months to come back and add the `dateUpdated` field because some client decided they wanted it.